in_toto/verifylib.py

```diff
@@ -1039,45 +1040,142 @@ def reduce_chain_links(chain_link_dict):
+
+  <Arguments>
+    layout:
+            The layout specified by the project owner.
+
+    reduced_chain_link_dict:
+            A dictionary of step-link pair with step names as keys.
+
+  <Exceptions>
+    None.
+
+  <Side Effects>
+    None.
+
+  <Returns>
+    A link which summarizes the materials and products of the overall
+    software supply chain.
+  """
+
+  # Create a new link object to return
+  summary_link = in_toto.models.link.Link()
+
+  first_step_link = reduced_chain_link_dict[layout.steps[0].name]
+  last_step_link = reduced_chain_link_dict[layout.steps[-1].name]
+
+  summary_link.materials = first_step_link.materials
+  summary_link._type = first_step_link._type
+  summary_link.name = first_step_link.name
+
+  summary_link.products = last_step_link.products
+  summary_link.byproducts = last_step_link.byproducts
+  summary_link.command = last_step_link.command
+  summary_link.return_value = last_step_link.return_value
+
+  return summary_link
+
+def in_toto_verify(layout, layout_key_dict):
   """
   <Purpose>
     Does entire in-toto supply chain verification of a final product
     by performing the following actions:
-        1.  Load root layout as specified by 'layout_path'
-        2.  Load root layout signing key(s) (project owner public key(s)) as
-            specified by 'layout_key_paths'
-        3.  Verify layout signature(s)
-        4.  Verify layout expiration
-        5.  Load link metadata for every Step defined in the layout
+        1.  Verify layout signature(s)
+        2.  Verify layout expiration
+        3.  Load link metadata for every Step defined in the layout
             NOTE: in_toto_verify searches for link metadata files by their step
             name plus the extension '.link' in the current directory.
             E.g. a step of name 'foo' is expected to be found in the current
             directory at 'foo.link'.
-        6.  Verify functionary signature for every Link.
-        7.  Verify alignment of defined (Step) and reported (Link) commands
+        4.  Verify functionary signature for every Link.
+        5.  Verify sublayouts, if any.
+            NOTE: Replaces the layout object in the chain_link_dict with an
+            unsigned summary link. So, link signatures are verified before
+            this step and command alignments, thresholds and inspections are
+            verified later on.
+        6.  Verify alignment of defined (Step) and reported (Link) commands
             NOTE: Won't raise exception on mismatch
-        8.  Execute Inspection commands
+        7.  Execute Inspection commands
             NOTE: Inspections, similar to Steps executed with 'in-toto-run',
             will record materials before and products after command execution.
             For now it records everything in the current working directory.
-        9.  Verify rules defined in each Step's material_matchrules and
+        8.  Verify rules defined in each Step's material_matchrules and
```

Comment:
You're missing the threshold verification step here.

Btw. I think "verify threshold equality" is not an ideal name, because you are not verifying the equality of thresholds but rather "link equality for a step with threshold greater one". You don't have to fix this here, but you could write a ticket and/or PR.